### PR TITLE
Fix some typos

### PR DESCRIPTION
**Pull request by Eijebong**
I wrote a really naive script and found those typos in the documentation.

### CODE REVIEW

File: src/libcore/num/dec2flt/algorithm.rs [hunk 1]
```diff
@@ -336,7 +336,7 @@ pub fn algorithm_m<T: RawFloat>(f: &Big, e: i16) -> T {
     round_by_remainder(v, rem, q, z)
 }
 
-/// Skip over most AlgorithmM iterations by checking the bit length.
+/// Skip over most Algorithm M iterations by checking the bit length.
```
Comment (lukaramu) [inline src/libcore/num/dec2flt/algorithm.rs]: I think this should be "algorithm", not "Algorithm M"

**Review by chordowl — COMMENTED** [review]
These are quite amazing! :heart:

File: src/libcore/num/dec2flt/algorithm.rs (hunk 1 above)
Comment (Eijebong) [inline src/libcore/num/dec2flt/algorithm.rs]: There is another `Algorithm M` in the file, I assumed it was a name for an algorithm that was implemented in the file...

**Review by steveklabnik — APPROVED** [review]

File: src/libcore/num/dec2flt/algorithm.rs (hunk 1 above)
Comment (steveklabnik) [inline src/libcore/num/dec2flt/algorithm.rs]: r=me after we sort this out

Comment (Eijebong) [inline src/libcore/num/dec2flt/algorithm.rs]: Should I change both to `algorithm` then ?

Comment (Eijebong) [inline src/libcore/num/dec2flt/algorithm.rs]: Nah, I just looked it up, it's documenting a function called `algorithm_m`

Comment (lukaramu) [inline src/libcore/num/dec2flt/algorithm.rs]: oops, my bad!

**Pull request merged**